### PR TITLE
Rough outline on how to prevent movement

### PR DESCRIPTION
**Pull request by policiu**
Address #185.

Let's use events to let Player do whatever it wants without impacting Commands. We use the current velocity direction for movement, so it acts as if the player is holding that direction w/o affecting other player inputs.

### CODE REVIEW

**Comment by valkyrienyanko** [comment]
Use of the `EventManager.cs` also could have been used here. Using a mix of this and the event manager bugs me, I would rather use one or the other. I am not sure which one is better, I know this was discussed before but I am still not satisfied with the answers given.

**Comment by policiu** [comment]
Unfortunately, here we really can't use the EventManager.cs without heavily modifying it since that is a more of a global event system. I wouldn't think we want to have a bunch of things listening in on each command Event and having to decide if the given event applies.  EventManager is nice for things that are general (net update, global pause, global clean up, etc.) and non-specific while the command Events are for each instantiation of a command since each command doesn't call each other's events.

In other words, EventManager to me seems like a broadcast "Hey everyone, this happened!" (Command -> all) while the events in the Commands are more intimate "Hey those who know about me, this happened!" (Command -> single entity). 

EventManager could be changed to also use C# events, but that still doesn't really change the fact that EventManager is a global events system.

Open to discussion.

**Comment by valkyrienyanko** [comment]
I see so it's like dependency injection keeping everything contained.
